app/lib/features/chat_ng/utils.dart

```diff
@@ -63,3 +65,16 @@ class ChatEditorUtils {
   /// max height for the editor
   static const double maxHeight = 200.0;
 }
+
+/// format a timestamp to either 12-hour or 24-hour format based on device settings.
+String jiffyMsgTimestamp(BuildContext context, int timeInterval) {
```

Comment:
why not make this a widget? any function that takes a `BuildContext` and returns something to render should clearly be a widget class ;) .